wallet/src/repository/wallet/mod.rs

```diff
@@ -373,7 +373,34 @@ where
         let mut addresses = Vec::with_capacity(range.len());
 
         log::debug!(
-            "Retrieving addresses in range {:?}. Start({}), End({}), Total({})",
+            "Retrieving external addresses in range {:?}. Start({}), End({}), Total({})",
+            range,
+            start,
+            end,
+            total
+        );
+        for index in range.rev() {
+            let address = self.get_address(account, keychain, index)?;
+            addresses.push((*address).clone());
+        }
+
+        Ok(model::Addresses { addresses, total })
+    }
+
+    /// Return a list of the generated internal addresses that.
+    pub fn internal_addresses(&self, offset: u32, limit: u32) -> Result<model::Addresses> {
```

Comment:
I think this function can be refactored together with `external_addresses`.

Maybe creating a generic non-public `addresses` function with the `limit`, `offset` and `keychain` as argument.